src/safeposix/syscalls/sys_calls.rs

```diff
@@ -69,15 +69,49 @@ impl Cage {
         }
     }
 
+    /// ### Description
+    /// 
+    ///'fork_syscall` creates a new process (cage object)
+    /// The newly created child process is an exact copy of the 
+    /// parent process (the process that calls fork) 
+    /// apart from it's cage_id and the parent_id
+    /// In this function we clone the mutex table, condition variables table, 
+    /// semaphore table and the file descriptors and create 
+    /// a new Cage object with these cloned tables. 
+    /// We also update the shared memory mappings - and create mappings 
+    /// from the new Cage object the the 
+    /// parent Cage object's memory mappings. 
+    /// 
+    /// ### Arguments
+    /// 
+    /// It accepts one parameter:
+    /// 
+    /// * `child_cageid` : an integer representing the pid of the child process
+    /// 
+    /// ### Errors
+    ///    
+    /// There are 2 scenarios where the call to `fork_syscall` might return an error
+    /// 
+    /// * When the RawMutex::create() call fails to create a new Mutex object
+    /// * When the RawCondvar::create() call fails to create a new Condition Variable object
+    /// 
+    /// ### Returns
+    /// 
+    /// On success it returns a value of 0, and the new child Cage object is added to Cagetable
```

Comment:
Please add a `Panic` section as well to follow the standard notation with all syscalls. If we have any panics, we can mention them otherwise just mentioning that we don't have any panics also works

Reply:
Please add a reference to the manpage for `fork` call as well.